### PR TITLE
List responses can be shown now

### LINKED ISSUE #1236: In list responses "An error occurred" is displayed
**Actual Behaviour**

Currently whenever we request any answer whose return will be a list, susi displays "An error occurred", instead of a list.

**Expected Behaviour**

A proper list should be displayed.

**Steps to reproduce it**

Open susi app and in the chatActivity type "Barcelona Team players", the response from susi will be  "An error occurred"

**Screenshot for the issue**

screenshot_20180515-083150

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1236 

Changes: List response can be accommodated now in the app.

Screenshots for the change: 
screenshot_20180515-160717

### CODE REVIEW

**Comment by batbrain7** [comment]
@chiragw15 @arundhati24 please review

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/Datum.java [hunk 1]
```diff
@@ -49,12 +49,48 @@
...
+        this.position = position;
+    }
 
     /**
      * Gets 0.
      *
      * @return the 0
      */
+
+
```
[the remark this answers is not in the document]
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/Datum.java]: Rest looks good to me.

[the remark this answers is not in the document]
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/Datum.java]: Committing the changes @arundhati24

File: app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt [hunk 2]
```diff
@@ -60,10 +60,19 @@ class ParseSusiResponseHelper {
 
             Constant.RSS -> try {
                 datumList = susiResponse.answers[0].data
+                for (datum in datumList!!) {
+
+                }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt]: Why this empty loop?

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java [hunk 3]
```diff
@@ -190,6 +191,11 @@ public void onDetachedFromRecyclerView(RecyclerView recyclerView) {
                 SearchResultsListHolder searchResultsListHolder = new SearchResultsListHolder(view);
                 searchResultsListHolder.recyclerView.addItemDecoration(new ConstraintsHelper(6, currContext));
                 return searchResultsListHolder;
+            case LIST_RESULT :
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/ChatFeedRecyclerAdapter.java]: Remove space before `:`

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 4]
```diff
@@ -43,13 +43,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final RssViewHolder holder, int position) {
         Datum datum = datumList.get(position);
         if (datum != null) {
-            if( datum.getTitle() != null && !datum.getTitle().isEmpty()) {
-                holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
+            if(datum.getLink()!=null && !datum.getLink().isEmpty()) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Space around `!=`

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 5]
```diff
@@ -43,13 +43,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+                if( datum.getTitle() != null && !datum.getTitle().isEmpty()) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Remove space before `datum`

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Just use `TextUtils.isEmpty`

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: TextUtils.isEmpty

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Space after if

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java (hunk 5 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Space after if

**Comment by batbrain7** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 6]
```diff
@@ -43,13 +44,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+                }
+                holder.linkTextView.setText(datum.getLink());
+            } else {
+                if (TextUtils.isEmpty(datum.getName())) {
+                    holder.titleTextView.setText(Html.fromHtml(datum.getName()));
+                }
+                if (TextUtils.isEmpty(datum.getJerseyNumber())) {
+                    holder.descriptionTextView.setText(Html.fromHtml(datum.getJerseyNumber()));
+                }
+                if (datum.getPosition()!= null && !datum.getPosition().isEmpty()) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: TextUtils

**Comment by batbrain7** [comment]
@iamareebjamal please review now

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 7]
```diff
@@ -43,13 +44,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final RssViewHolder holder, int position) {
         Datum datum = datumList.get(position);
         if (datum != null) {
-            if( datum.getTitle() != null && !datum.getTitle().isEmpty()) {
-                holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
+            if(TextUtils.isEmpty(datum.getLink())) {
+                if (TextUtils.isEmpty(datum.getTitle())) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Opposite logic

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 8]
```diff
@@ -43,13 +44,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     public void onBindViewHolder(final RssViewHolder holder, int position) {
         Datum datum = datumList.get(position);
         if (datum != null) {
-            if( datum.getTitle() != null && !datum.getTitle().isEmpty()) {
-                holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
+            if(TextUtils.isEmpty(datum.getLink())) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Space after if

File: app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java [hunk 9]
```diff
@@ -43,13 +44,25 @@ public RssViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+            } else {
+                if (TextUtils.isEmpty(datum.getName())) {
+                    holder.titleTextView.setText(Html.fromHtml(datum.getName()));
+                }
+                if (TextUtils.isEmpty(datum.getJerseyNumber())) {
+                    holder.descriptionTextView.setText(Html.fromHtml(datum.getJerseyNumber()));
+                }
+                if (TextUtils.isEmpty(datum.getPosition())) {
+                    holder.linkTextView.setText(Html.fromHtml(datum.getPosition()));
+                }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: Your logic is opposite in all these cases

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/chat/adapters/recycleradapters/SearchResultsAdapter.java]: @iamareebjamal what does opposite logic mean?

**Comment by iamareebjamal** [comment]
You want to show text when it is not empty

**Comment by batbrain7** [comment]
@iamareebjamal please review

**Pull request merged**